pkg/collector/corechecks/embed/jmx/state.go

```diff
@@ -0,0 +1,67 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+// +build jmx
+
+package jmx
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/autodiscovery/integration"
+	"github.com/DataDog/datadog-agent/pkg/collector/check"
+	"github.com/DataDog/datadog-agent/pkg/util/cache"
+)
+
+type jmxState struct {
+	configs     *cache.BasicCache
+	runnerError chan struct{}
+	runner      *runner
+	lock        *sync.Mutex
+}
+
+var state jmxState = jmxState{
+	configs:     cache.NewBasicCache(),
+	runnerError: make(chan struct{}),
+	runner:      &runner{},
+	lock:        &sync.Mutex{},
+}
+
+func (s *jmxState) scheduleCheck(c *JMXCheck) error {
+	s.lock.Lock()
+	defer s.lock.Unlock()
+	if !s.runner.started {
+		err := check.Retry(5*time.Second, 3, s.runner.startRunner, "jmxfetch")
+		if err != nil {
+			return err
+		}
+	}
+	s.configs.Add(string(c.id), c.config)
+	return nil
+}
+
+func (s *jmxState) unscheduleCheck(c *JMXCheck) {
+	s.lock.Lock()
+	defer s.lock.Unlock()
+	s.configs.Remove(string(c.id))
+}
+
+func AddScheduledConfig(c integration.Config) {
```

Comment:
You're still missing the comments for these exported functions, could you add them for `AddScheduledConfig` and `GetScheduledConfig` with a brief description?